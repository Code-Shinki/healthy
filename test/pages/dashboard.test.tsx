import { cleanup, render, screen } from '@testing-library/react'
import { useRouter } from 'next/router'
import React from 'react'
import Dashboard from '../../src/pages/dashboard'
import AppMock from '../mocks/AppMock'
import { loggedInDemo, loggedInInit, nowLoading, unregistered } from '../mocks/globalStateMock'
import { nextRouterMock } from '../mocks/nextRouterMock'

// モック関数を作成
jest.mock('next/router', () => ({ useRouter: jest.fn() }))
jest.mock('react-apexcharts', () => {
  return {
    __esModule: true,
    // eslint-disable-next-line react/display-name
    default: () => {
      return <div />
    },
  }
})

describe(`ダッシュボード (/dashboard)`, () => {
  afterEach(cleanup)

  test('スナップショット', async () => {
    // useRouterのメソッドをモック化する
    const mockRouter = nextRouterMock
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    const { asFragment } = render(
      <AppMock initializeState={loggedInDemo} drawerMenu={true}>
        <Dashboard />
      </AppMock>
    )

    const tree = asFragment()
    expect(tree).toMatchSnapshot()

    // 再レンダーを待機
    expect(await screen.findByText('最近の調子')).toBeInTheDocument()
  })

  test('ログイン状態を判別中', async () => {
    // ソースファイルのrouterに渡した引数をmockRouterで受け取る
    const mockRouter = nextRouterMock
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    render(
      <AppMock initializeState={nowLoading} drawerMenu={true}>
        <Dashboard />
      </AppMock>
    )

    // 読み込み中にcomponents/atoms/spinner.tsxが表示されるか
    expect(screen.getByText('読み込んでいます')).toBeInTheDocument()
  })

  test('ログインしていないユーザー', async () => {
    const mockRouter = nextRouterMock
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    render(
      <AppMock initializeState={unregistered} drawerMenu={true}>
        <Dashboard />
      </AppMock>
    )

    // next/routerでログインページへリダイレクトしているか
    expect(mockRouter.push).toHaveBeenCalledWith('/login')
  })

  test('初期データを持つログイン済みのユーザー', async () => {
    const mockRouter = nextRouterMock
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    render(
      <AppMock initializeState={loggedInInit} drawerMenu={true}>
        <Dashboard />
      </AppMock>
    )

    // メニューが存在するか
    expect(screen.getByRole('button', { name: 'ホーム' })).toBeInTheDocument()

    // 各コンテナのタイトルが存在するか
    expect(screen.getAllByRole('heading', { level: 2 })).toHaveLength(4)

    // データが規定数以上存在しないため注意が表示されるか
    expect(screen.getAllByText('データが不足しています')).toHaveLength(3)
  })

  test('フルデータを持つログイン済みのユーザー', async () => {
    // useRouterのメソッドをモック化する
    const mockRouter = nextRouterMock
    ;(useRouter as jest.Mock).mockReturnValue(mockRouter)

    render(
      <AppMock initializeState={loggedInDemo} drawerMenu={true}>
        <Dashboard />
      </AppMock>
    )

    // メニューが存在するか
    expect(screen.getByRole('button', { name: 'ホーム' })).toBeInTheDocument()

    // サマリーデータが存在するか
    expect(screen.getByRole('heading', { level: 2, name: '1月1日の体調' })).toBeInTheDocument()
    expect(screen.getByText('寝不足')).toBeInTheDocument()
    expect(screen.getByText('今日の体調を記録しましょう！')).toBeInTheDocument()

    // コンディションリストが存在するか
    expect(screen.getByRole('heading', { level: 2, name: '最近の調子' })).toBeInTheDocument()

    // 体温グラフが存在するか
    expect(screen.getByRole('heading', { level: 2, name: '最近の体温' })).toBeInTheDocument()
    expect(await screen.findByRole('region')).toHaveClass('miniGraph')

    // ユーザー情報が存在するか
    expect(screen.getByRole('heading', { level: 2, name: 'ユーザー情報' })).toBeInTheDocument()
    expect(screen.getByText('山本 太郎')).toBeInTheDocument()
    expect(screen.getByText('男性')).toBeInTheDocument()
    expect(screen.getByText('172 cm')).toBeInTheDocument()
    expect(screen.getByText('65.4 kg')).toBeInTheDocument()
    expect(screen.getByText('樫山市立中央病院')).toBeInTheDocument()
  })
})
