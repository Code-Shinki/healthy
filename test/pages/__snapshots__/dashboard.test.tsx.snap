// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ダッシュボード (/dashboard) スナップショット 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-1"
  >
    <div
      class="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
    >
      <div
        class="MuiGrid-root makeStyles-container-3 makeStyles-summary-4 MuiGrid-item MuiGrid-grid-xl-3"
      >
        <div
          class="root"
        >
          <div
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
              />
            </svg>
          </div>
          <h2
            class="h2"
          >
            1月1日の体調
          </h2>
          <div
            class="MuiGrid-root makeStyles-container-6 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-content-xs-center MuiGrid-justify-xs-center MuiGrid-grid-xs-4"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root good"
                focusable="false"
                style="font-size: 4.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
                />
              </svg>
              <div
                class="makeStyles-temperature-7"
              >
                36.5 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-around MuiGrid-grid-xs-8"
            >
              <div
                class="makeStyles-tag-8"
              >
                <div
                  class="tag"
                >
                  寝不足
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
          >
            <a
              class="makeStyles-link-9"
              href="/checkup"
            >
              今日の体調を記録しましょう！
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-container-3 MuiGrid-item MuiGrid-grid-xl-7"
      >
        <div
          class="root"
        >
          <div
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 10H7v2h10v-2zm2-7h-1V1h-2v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H5V8h14v11zm-5-5H7v2h7v-2z"
              />
            </svg>
          </div>
          <h2
            class="h2"
          >
            最近の調子
          </h2>
          <div
            class="MuiGrid-root makeStyles-container-10 MuiGrid-container MuiGrid-justify-xs-space-around"
          >
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root bad"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 13.5c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5zM7.82 12l1.06-1.06L9.94 12 11 10.94 9.94 9.88 11 8.82 9.94 7.76 8.88 8.82 7.82 7.76 6.76 8.82l1.06 1.06-1.06 1.06zm4.17-10C6.47 2 2 6.47 2 12s4.47 10 9.99 10S22 17.53 22 12 17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4.18-12.24l-1.06 1.06-1.06-1.06L13 8.82l1.06 1.06L13 10.94 14.06 12l1.06-1.06L16.18 12l1.06-1.06-1.06-1.06 1.06-1.06z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                37.6 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root bad"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 13.5c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5zM7.82 12l1.06-1.06L9.94 12 11 10.94 9.94 9.88 11 8.82 9.94 7.76 8.88 8.82 7.82 7.76 6.76 8.82l1.06 1.06-1.06 1.06zm4.17-10C6.47 2 2 6.47 2 12s4.47 10 9.99 10S22 17.53 22 12 17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4.18-12.24l-1.06 1.06-1.06-1.06L13 8.82l1.06 1.06L13 10.94 14.06 12l1.06-1.06L16.18 12l1.06-1.06-1.06-1.06 1.06-1.06z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                38.5 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root bad"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 13.5c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5zM7.82 12l1.06-1.06L9.94 12 11 10.94 9.94 9.88 11 8.82 9.94 7.76 8.88 8.82 7.82 7.76 6.76 8.82l1.06 1.06-1.06 1.06zm4.17-10C6.47 2 2 6.47 2 12s4.47 10 9.99 10S22 17.53 22 12 17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4.18-12.24l-1.06 1.06-1.06-1.06L13 8.82l1.06 1.06L13 10.94 14.06 12l1.06-1.06L16.18 12l1.06-1.06-1.06-1.06 1.06-1.06z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                37.8 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root bad"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 13.5c-2.33 0-4.31 1.46-5.11 3.5h10.22c-.8-2.04-2.78-3.5-5.11-3.5zM7.82 12l1.06-1.06L9.94 12 11 10.94 9.94 9.88 11 8.82 9.94 7.76 8.88 8.82 7.82 7.76 6.76 8.82l1.06 1.06-1.06 1.06zm4.17-10C6.47 2 2 6.47 2 12s4.47 10 9.99 10S22 17.53 22 12 17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm4.18-12.24l-1.06 1.06-1.06-1.06L13 8.82l1.06 1.06L13 10.94 14.06 12l1.06-1.06L16.18 12l1.06-1.06-1.06-1.06 1.06-1.06z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                37.2 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root fine"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5"
                />
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-3.5c.73 0 1.39.19 1.97.53.12-.14.86-.98 1.01-1.14-.85-.56-1.87-.89-2.98-.89-1.11 0-2.13.33-2.99.88.97 1.09.01.02 1.01 1.14.59-.33 1.25-.52 1.98-.52z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                36.8 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root good"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                36.4 ℃
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-11 MuiGrid-item"
            >
              <div
                class="makeStyles-date-12"
              >
                1日
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root good"
                focusable="false"
                style="font-size: 3.5em;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
                />
              </svg>
              <div
                class="makeStyles-temperature-13"
              >
                36.5 ℃
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-container-3 MuiGrid-item MuiGrid-grid-xl-7"
      >
        <div
          class="root"
        >
          <div
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z"
              />
            </svg>
          </div>
          <h2
            class="h2"
          >
            最近の体温
          </h2>
          <section
            aria-label="graph-wrapper"
            class="miniGraph"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-container-3 makeStyles-user-5 MuiGrid-item MuiGrid-grid-xl-3"
      >
        <div
          class="root"
        >
          <div
            class="icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <h2
            class="h2"
          >
            ユーザー情報
          </h2>
          <div
            class="makeStyles-container-14"
          >
            <h3
              class="label"
            >
              ユーザー名
            </h3>
            <p
              class="makeStyles-data-15"
            >
              山本 太郎
            </p>
            <h3
              class="label"
            >
              性別
            </h3>
            <p
              class="makeStyles-data-15"
            >
              男性
            </p>
            <h3
              class="label"
            >
              身長
            </h3>
            <p
              class="makeStyles-data-15"
            >
              172 cm
            </p>
            <h3
              class="label"
            >
              体重
            </h3>
            <p
              class="makeStyles-data-15"
            >
              65.4 kg
            </p>
            <h3
              class="label"
            >
              かかりつけ医
            </h3>
            <p
              class="makeStyles-data-15"
            >
              樫山市立中央病院
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiBottomNavigation-root root"
  >
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root item"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          ホーム
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root item"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          ヘルスデータ
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root item"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          体温グラフ
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root item"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          ユーザー
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiBottomNavigationAction-root item"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiBottomNavigationAction-label"
        >
          ログアウト
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
